### PR TITLE
Enable Pallet XCM for Kusama & Westend

### PR DESCRIPTION
**Pull request by gavofyork**
not-live because it doesn't touch Polkadot runtime.

### CODE REVIEW

**Review by shawntabrizi — APPROVED** [review]

File: runtime/kusama/src/lib.rs [hunk 1]
```diff
@@ -1146,6 +1146,79 @@ impl xcm_executor::Config for XcmConfig {
...
+impl frame_support::traits::Contains<(MultiLocation, Xcm<Call>)> for OnlyWithdrawTeleportForAccounts {
+	fn contains((ref origin, ref msg): &(MultiLocation, Xcm<Call>)) -> bool {
+		use xcm::v0::{
+			Xcm::WithdrawAsset, Order::{BuyExecution, InitiateTeleport, DepositAsset},
+			MultiAsset::{All, ConcreteFungible}, Junction::{AccountId32, Plurality, Parachain},
+			MultiLocation::{Null, X1},
+		};
+		match origin {
+			// Root and council are are allowed to execute anything.
+			Null | X1(Plurality { .. }) => true,
```
Comment (shawntabrizi) [inline runtime/kusama/src/lib.rs:1173]: We don't do any verification on the `Plurality`. Seems fine now with only council, but could be a foot gun later.

Comment (gavofyork) [reply runtime/kusama/src/lib.rs:1173]: i'm assumign that collectives in general don't need the same safeguards as random accounts (which are trivially sybillable).

Comment (shawntabrizi) [reply runtime/kusama/src/lib.rs:1173]: actually we allow any 1 / N council origin to pass here?

**Review by shawntabrizi — CHANGES_REQUESTED** [review]
I think 1 / N council here is given full access to XCM

**Comment by gavofyork** [comment]
yeah. that's fine.

**Review by shawntabrizi — APPROVED** [review]
Note to make sure that statemint or any downstream parachain correctly checks the threshold of the plurality origin

cc @joepetrowski

**Review by andresilva — APPROVED** [review]

File: runtime/westend/src/lib.rs [hunk 2]
```diff
@@ -878,6 +879,72 @@ impl xcm_executor::Config for XcmConfig {
...
+impl frame_support::traits::Contains<(MultiLocation, Xcm<Call>)> for OnlyWithdrawTeleportForAccounts {
+	fn contains((ref origin, ref msg): &(MultiLocation, Xcm<Call>)) -> bool {
+		use xcm::v0::{
+			Xcm::WithdrawAsset, Order::{BuyExecution, InitiateTeleport, DepositAsset},
+			MultiAsset::{All, ConcreteFungible}, Junction::{AccountId32, Parachain},
+			MultiLocation::{Null, X1},
+		};
+		match origin {
+			// Root is allowed to execute anything.
+			Null => true,
```
Comment (apopiak) [inline runtime/westend/src/lib.rs:899]: It seems that this will mean we cannot test plurality XCMs on Westend?

Comment (andresilva) [reply runtime/westend/src/lib.rs:899]: We don't have a council on Westend since we have sudo. Your point might still be relevant though.

**Comment by apopiak** [comment]
> Note to make sure that statemint or any downstream parachain correctly checks the threshold of the plurality origin
> 
> cc @joepetrowski

It will check `is_majority` for now, could check other percentages later.

File: runtime/westend/src/lib.rs (hunk 2 above)
Comment (gavofyork) [reply runtime/westend/src/lib.rs:879]: We could introduce a council on Westend, but I think it's probably enough to test on Rococo.

**Review by rphmeier — APPROVED** [review]

**Comment by andresilva** [comment]
bot merge

**Pull request merged**